### DISCUSSION TITLE
Creating custom environment

### DISCUSSION THREAD

**Discussion by anaskn** [PKU-Alignment/omnisafe#275] [General]
_**When creating a new custom environment and going through the following steps I get an error at custom_env.make:**_

![Screen Shot 2023-09-05 at 13 20 09](https://github.com/PKU-Alignment/omnisafe/assets/37979554/53e8422e-9d0d-4baa-8579-e76f59a4cbf5)

**AttributeError: type object 'custom_env' has no attribute 'make'**

**Comment by Gaiejj** [comment]
We apologize for any misunderstandings that may have arisen. The examples provided here are merely for reference purposes and are not intended to be directly executable code. Due to the diversity of the SafeRL environment interfaces, it is difficult for us to provide a comprehensive interface that supports all environments. Therefore, **the code blocks in the screenshot you provided are symbolic examples.**

If you need real examples of customized environments, we recommend referring to the [Safety-Gymnasium](https://github.com/PKU-Alignment/omnisafe/blob/main/omnisafe/envs/safety_gymnasium_env.py) and [MetaDrive](https://github.com/PKU-Alignment/omnisafe/pull/263) interfaces. These examples can provide you with more specific and realistic references for creating and working with customized environments. We hope this helps. If you have any further questions, please feel free to ask.
